Perl-Based GitHub Actions Failing

# Bug Report

## Problem and Scope
GitHub Actions that involve Perl are using [`shogo82148/actions-setup-perl@v1`](https://github.com/shogo82148/actions-setup-perl/tree/v1) internally for setting up Perl and installing packages from CPAN, currently in use for `Perl::Critic` and `Perl::Tidy` for linting and formatting respectively.

Currently any attempt to use the action fails with different attestation warnings / verification errors.

Relevant issues include
- https://github.com/shogo82148/actions-setup-perl/issues/2481 (primary)
- https://github.com/shogo82148/actions-setup-perl/issues/2429 (potentially unrelated)

After the original one was closed another was created
- https://github.com/shogo82148/actions-setup-perl/issues/2483

## Steps to Reproduce
1. Run [`shogo82148/actions-setup-perl@v1`](https://github.com/shogo82148/actions-setup-perl/tree/v1) on GitHub Actions

## Testing Environment
GitHub Actions

## Bug Discovery
```
2026-02-14T10:30:53.5290199Z ##[group]Run shogo82148/actions-setup-perl@v1
2026-02-14T10:30:53.5290913Z with:
2026-02-14T10:30:53.5291372Z   install-modules: Perl::Critic
2026-02-14T10:30:53.5292060Z   github-token: ***
2026-02-14T10:30:53.5292998Z   distribution: default
2026-02-14T10:30:53.5293494Z   enable-modules-cache: true
2026-02-14T10:30:53.5294000Z   working-directory: .
2026-02-14T10:30:53.5294420Z ##[endgroup]
2026-02-14T10:30:53.7189055Z ##[group]install perl
2026-02-14T10:30:54.2266470Z verifying download: /home/runner/work/_temp/e4f1cb87-4f40-41fc-9da3-fff0dc41e9da
2026-02-14T10:30:54.4818291Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-14T10:30:54.4866624Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-14T10:30:54.4870249Z ##[endgroup]
2026-02-14T10:30:54.4884562Z ##[error]Error: failed to verify /home/runner/work/_temp/e4f1cb87-4f40-41fc-9da3-fff0dc41e9da
```

and

```
2026-02-14T10:30:59.2719840Z ##[group]Run shogo82148/actions-setup-perl@v1
2026-02-14T10:30:59.2721296Z with:
2026-02-14T10:30:59.2722191Z   install-modules: Perl::Tidy
2026-02-14T10:30:59.2724046Z   github-token: ***
2026-02-14T10:30:59.2725115Z   distribution: default
2026-02-14T10:30:59.2726192Z   enable-modules-cache: true
2026-02-14T10:30:59.2727298Z   working-directory: .
2026-02-14T10:30:59.2728300Z ##[endgroup]
2026-02-14T10:30:59.6766910Z ##[group]install perl
2026-02-14T10:31:00.1663221Z verifying download: /home/runner/work/_temp/5b98e239-6d29-417e-bc85-c944ef482c5c
2026-02-14T10:31:00.3583897Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-14T10:31:00.3752282Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-14T10:31:00.3757572Z ##[endgroup]
2026-02-14T10:31:00.3762287Z ##[error]Error: failed to verify /home/runner/work/_temp/5b98e239-6d29-417e-bc85-c944ef482c5c
```

## Estimated Severity
Critical

Attempting to add more permissions in #220 failed with:

```
2026-02-16T00:08:27.9744199Z ##[group]Run shogo82148/actions-setup-perl@v1
2026-02-16T00:08:27.9744868Z with:
2026-02-16T00:08:27.9745365Z   install-modules: Perl::Critic
2026-02-16T00:08:27.9746065Z   github-token: ***
2026-02-16T00:08:27.9746610Z   distribution: default
2026-02-16T00:08:27.9747092Z   enable-modules-cache: true
2026-02-16T00:08:27.9747612Z   working-directory: .
2026-02-16T00:08:27.9748076Z ##[endgroup]
2026-02-16T00:08:28.6359048Z ##[group]install perl
2026-02-16T00:08:28.6572203Z verifying download: /home/runner/work/_temp/54fe4708-2b46-473c-8929-504a164d7783
2026-02-16T00:08:28.9507680Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-16T00:08:28.9544339Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-16T00:08:28.9547053Z ##[endgroup]
2026-02-16T00:08:28.9561288Z ##[error]Error: failed to verify /home/runner/work/_temp/54fe4708-2b46-473c-8929-504a164d7783
```

and

```
2026-02-16T00:08:41.0388844Z ##[group]Run shogo82148/actions-setup-perl@v1
2026-02-16T00:08:41.0390616Z with:
2026-02-16T00:08:41.0391560Z   install-modules: Perl::Tidy
2026-02-16T00:08:41.0393230Z   github-token: ***
2026-02-16T00:08:41.0394341Z   distribution: default
2026-02-16T00:08:41.0399188Z   enable-modules-cache: true
2026-02-16T00:08:41.0400777Z   working-directory: .
2026-02-16T00:08:41.0401858Z ##[endgroup]
2026-02-16T00:08:41.3407281Z ##[group]install perl
2026-02-16T00:08:41.8629417Z verifying download: /home/runner/work/_temp/1cdd00c6-9158-4a51-814f-271046f7d961
2026-02-16T00:08:42.0373555Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-16T00:08:42.0419435Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-16T00:08:42.0436273Z ##[endgroup]
2026-02-16T00:08:42.0454392Z ##[error]Error: failed to verify /home/runner/work/_temp/1cdd00c6-9158-4a51-814f-271046f7d961
```

Attempting to disable module caching in #220 failed with:

```
2026-02-16T00:20:04.6094796Z ##[group]Run shogo82148/actions-setup-perl@v1
2026-02-16T00:20:04.6096020Z with:
2026-02-16T00:20:04.6096963Z   install-modules: Perl::Critic
2026-02-16T00:20:04.6098373Z   enable-modules-cache: false
2026-02-16T00:20:04.6099805Z   github-token: ***
2026-02-16T00:20:04.6100778Z   distribution: default
2026-02-16T00:20:04.6101679Z   working-directory: .
2026-02-16T00:20:04.6102572Z ##[endgroup]
2026-02-16T00:20:04.8029341Z ##[group]install perl
2026-02-16T00:20:05.1208926Z verifying download: /home/runner/work/_temp/e6c928aa-41e6-4c11-a71c-6d84d5940924
2026-02-16T00:20:05.3384627Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-16T00:20:05.3479111Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-16T00:20:05.3486967Z ##[endgroup]
2026-02-16T00:20:05.3493248Z ##[error]Error: failed to verify /home/runner/work/_temp/e6c928aa-41e6-4c11-a71c-6d84d5940924
```

and

```
2026-02-16T00:20:05.4676457Z ##[group]Run shogo82148/actions-setup-perl@v1
2026-02-16T00:20:05.4677188Z with:
2026-02-16T00:20:05.4677692Z   install-modules: Perl::Tidy
2026-02-16T00:20:05.4678265Z   enable-modules-cache: false
2026-02-16T00:20:05.4679001Z   github-token: ***
2026-02-16T00:20:05.4679491Z   distribution: default
2026-02-16T00:20:05.4679996Z   working-directory: .
2026-02-16T00:20:05.4680490Z ##[endgroup]
2026-02-16T00:20:05.6492278Z ##[group]install perl
2026-02-16T00:20:06.1308389Z verifying download: /home/runner/work/_temp/9e473c15-5fec-461f-8b15-06e471e04e3c
2026-02-16T00:20:06.4391809Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-16T00:20:06.4413792Z ##[warning]failed to verify attestation: TypeError: Cannot read properties of null (reading 'mediaType'), continue to next attestation
2026-02-16T00:20:06.4426482Z ##[endgroup]
2026-02-16T00:20:06.4444264Z ##[error]Error: failed to verify /home/runner/work/_temp/9e473c15-5fec-461f-8b15-06e471e04e3c
```

--- a/.github/workflows/AutoFormat.yml
+++ b/.github/workflows/AutoFormat.yml
@@ -117,7 +117,7 @@ jobs:
           fetch-depth: 1
 
       - name: Setup Perl
-        uses: shogo82148/actions-setup-perl@v1.33.0
+        uses: shogo82148/actions-setup-perl@v1
         with:
           install-modules: 'Perl::Tidy'
 

--- a/.github/workflows/Linter.yml
+++ b/.github/workflows/Linter.yml
@@ -108,7 +108,7 @@ jobs:
           fetch-depth: 1
 
       - name: Setup Perl
-        uses: shogo82148/actions-setup-perl@v1.33.0
+        uses: shogo82148/actions-setup-perl@v1
         with:
           install-modules: 'Perl::Critic'
 

